mentions-generator/src/main/java/de/brandwatch/minianalytics/mentiongenerator/service/LuceneService.java

```diff
@@ -0,0 +1,131 @@
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+@Service
+public class LuceneService {
+
+    private static final Logger logger = LoggerFactory.getLogger(LuceneService.class);
+
+    private final Producer producer;
+
+    private final QueryCache queryCache;
+
+    private final Directory memoryIndex = new RAMDirectory();
+    private final StandardAnalyzer analyzer = new StandardAnalyzer();
+
+    private final IndexWriterConfig indexWriterConfig = new IndexWriterConfig(analyzer);
+    private final IndexWriter writer = new IndexWriter(memoryIndex, indexWriterConfig);
+
+    private final QueryParser queryParser = new QueryParser("text", analyzer);
+
+    private final ConcurrentLinkedQueue<Resource> queue = new ConcurrentLinkedQueue<>();
+
+    private IndexSearcher searcher;
+    private IndexReader reader;
+
+    @Autowired
+    public LuceneService(Producer producer, QueryCache queryCache) throws IOException {
+        this.producer = producer;
+        this.queryCache = queryCache;
+    }
+
+    public void writeToQ(Resource resource) {
+        queue.add(resource);
+    }
+
+    @Scheduled(fixedDelay = 5 * 1000)
+    public void indexQ() throws IOException, ParseException {
+        while (!queue.isEmpty()) {
+            Resource resource = queue.poll();
+
+            Document document = new Document();
+
+            document.add(new TextField("author", resource.getAuthor(), Field.Store.YES));
+            document.add(new TextField("text", resource.getText(), Field.Store.YES));
+            document.add(new TextField("date", resource.getDate().toString(), Field.Store.YES));
+
+            writer.addDocument(document);
+            writer.commit();
+        }
+
+        searchIndex();
+    }
+
+    private void searchIndex() throws IOException, ParseException {
+
+        reader = DirectoryReader.open(memoryIndex);
+        searcher = new IndexSearcher(reader);
+
+        List<Query> queries = queryCache.getCachedQueries();
+        logger.info("received " + queries.size() + " queries");
+
+        for (Query query : queries) {
+
+            org.apache.lucene.search.Query luceneQuery = queryParser.parse(query.toString());
+
+            logger.info("Search Query: " + luceneQuery.toString());
+            //The maximum of hits can only be 1 bcs. there is a maximum of 1 Document in the index
+            TopDocs topDocs = searcher.search(luceneQuery, 10);
+
+            logger.info("Total hits: " + topDocs.totalHits);
+
+            if (topDocs.totalHits >= 1) {
+                Arrays.stream(topDocs.scoreDocs).map(doc -> {
+                    try {
+                        return toMention(doc, query);
+                    } catch (IOException e) {
+                        logger.warn(e.getMessage(), e);
```

Comment:
This should be error level. Also the message should be part of the e anyway, no? Isn't it duplicated like this? You could rather add additional useful information as a string.

Reply:
Yes, the message is part of the `e` anyways, but I cant just only print the `e`, there has to be a `String` in front.

Reply:
I think duplication is worse than something like `logger.error("An exception occurred during mention creation", e);`

Reply:
Ok, I can do that.